src/main/java/org/testng/internal/Parameters.java

```diff
@@ -102,10 +102,14 @@
 
     mapping.put(BeforeMethod.class.getSimpleName(), beforeAfterMethod);
     mapping.put(AfterMethod.class.getSimpleName(), beforeAfterMethod);
-    mapping.put(Test.class.getSimpleName(), Arrays.asList(new Class<?>[] {ITestContext.class}));
+    mapping.put(Test.class.getSimpleName(), newList(ITestContext.class));
 
   }
 
+  private static List<Class<?>> newList(Class<?>...elements) {
```

Comment:
@juherr... nice ! I never knew we could use `Arrays.asList()` in that fashion wherein we specify explicitly the return type... (Guess its time for me to go back and read more about `generics`). Yes, that would work. But now that brings me back to the question...

Apart from the succinctness (which is the only thing that stands obvious to my naive eyes), how is what you suggested different from what I have as an implementation in `newList()` ? I am curious. Can you please tell me ?